File: backend/Concord/Concord.Application/Models/SessionResponse.cs
using System.Text.Json.Serialization;

namespace Concord.Application.Models;

public class SessionResponse
{
    [JsonPropertyName("Id")]
    public Guid Id { get; set; }

    [JsonPropertyName("DeviceLabel")]
    public string? DeviceLabel { get; set; }

    [JsonPropertyName("Browser")]
    public string? Browser { get; set; }

    [JsonPropertyName("OS")]
    public string? OS { get; set; }

    [JsonPropertyName("IpAddress")]
    public string? IpAddress { get; set; }

    [JsonPropertyName("Created")]
    public DateTime Created { get; set; }

    [JsonPropertyName("LastActiveAt")]
    public DateTime LastActiveAt { get; set; }

    [JsonPropertyName("IsCurrent")]
    public bool IsCurrent { get; set; }
}

File: mobile/concord/lib/api/models/session_response.dart
import 'json_utils.dart';

class SessionResponse {
  const SessionResponse({
    required this.id,
    required this.deviceLabel,
    required this.browser,
    required this.os,
    required this.ipAddress,
    required this.created,
    required this.lastActiveAt,
    required this.isCurrent,
  });

  factory SessionResponse.fromJson(Map<String, dynamic> json) {
    return SessionResponse(
      id: json.field('Id') as String,
      deviceLabel: json.field('DeviceLabel') as String?,
      browser: json.field('Browser') as String?,
      os: json.field('OS') as String?,
      ipAddress: json.field('IpAddress') as String?,
      created: parseDateTime(json.field('Created')),
      lastActiveAt: parseDateTime(json.field('LastActiveAt')),
      isCurrent: json.field('IsCurrent') as bool,
    );
  }

  final String id;
  final String? deviceLabel;
  final String? browser;
  final String? os;
  final String? ipAddress;
  final DateTime created;
  final DateTime lastActiveAt;
  final bool isCurrent;
}
